packages/apollo-datasource-rest/src/__tests__/RESTDataSource.test.ts

```diff
@@ -779,4 +803,69 @@ describe('RESTDataSource', () => {
       );
     });
   });
+
+  describe('http cache', () => {
+    it('allows setting cache options for each request', async () => {
+      const dataSource = new (class extends RESTDataSource {
+        override baseURL = 'https://api.example.com';
+        override requestCacheEnabled = false;
+
+        getFoo(id: number) {
+          return this.get(`foo/${id}`);
+        }
+
+        // Set a long TTL for every request
+        // @ts-ignore
+        override cacheOptionsFor(_, __): CacheOptions | undefined {
```

Comment:
Just a nit, but I'd suggest either of these in favor of the @ts-ignore

```suggestion
        // Set a long TTL for every request
        override cacheOptionsFor(): CacheOptions | undefined {
```

```suggestion
        // Set a long TTL for every request
        override cacheOptionsFor(_res: Response, _req: Request): CacheOptions | undefined {
```